wandb/wandb_config.py

```diff
@@ -49,19 +50,30 @@ def __init__(self, config_paths=[], wandb_dir=None, run_dir=None):
         self._set_wandb('python_version', platform.python_version())
         self._set_wandb('is_jupyter_run', wandb._get_python_type() != "python")
 
+        # Detect frameworks
+        _fastai = sys.modules.get('fastai')
+        _torch = sys.modules.get('torch')
+        _keras = sys.modules.get('keras')
+        _tfkeras = sys.modules.get('tensorflow.python.keras')
+        _tensorflow = sys.modules.get('tensorflow')
+        framework = 'fastai' if _fastai else 'torch' if _torch else 'keras' if _keras else 'tfkeras' if _tfkeras else 'tensorflow' if _tensorflow else None
```

Comment:
`framework = ([f for f in mods if sys.modules.get(f)] + [None])[0]` could be cleaner, although this logic will likely get more complex.